File: sdk/java-sdk/src/main/java/com/evernym/verity/sdk/handlers/DefaultMessageHandler.java
package com.evernym.verity.sdk.handlers;

import org.json.JSONObject;

/**
 * Defines how to handle a message of a certain type and optionally with a particular status
 */
public class DefaultMessageHandler {
    private final Handler messageHandler;

    /**
     * A simple default handler interface for JSON object messages
     */
    public interface Handler {
        /**
         * Handles the given JSON object message
         * @param message the message to be handled
         */
        void handle(JSONObject message);
    }

    /**
     * Associates a handler callback function with the default handler action
     * @param messageHandler the handler function itself
     */
    DefaultMessageHandler(Handler messageHandler) {
        this.messageHandler = messageHandler;
    }

    /**
     * Calls the handler callback function for the given message
     * @param message the JSON structure of the agent message
     */
    public void handle(JSONObject message) {
        this.messageHandler.handle(message);
    }
}

File: sdk/dotnet-sdk/src/VeritySDK/Handlers/DefaultMessageHandler.cs
using System.Json;

namespace VeritySDK
{
    /// <summary>
    /// Defines how to handle a message of a certain type and optionally with a particular status
    /// </summary>
    public class DefaultMessageHandler
    {
        /// <summary>
        /// A simple default delegate handler interface for JSON object messages
        /// </summary>
        /// <param name="message">the JSON structure of the agent message</param>
        public delegate void Handler(JsonObject message);

        private Handler messageHandler;

        /// <summary>
        /// Associates a handler callback function with the default handler action
        /// </summary>
        /// <param name="messageHandler">the handler function itself</param>
        public DefaultMessageHandler(Handler messageHandler)
        {
            this.messageHandler = messageHandler;
        }

        /// <summary>
        /// Calls the handler callback function for the given message         
        /// </summary>
        /// <param name="message">the JSON structure of the agent message</param>
        public void handle(JsonObject message)
        {
            this.messageHandler(message);
        }
    }
}
